pymongo/topology.py

```diff
@@ -282,11 +282,13 @@ def _select_server(
         selector: Callable[[Selection], Selection],
         server_selection_timeout: Optional[float] = None,
         address: Optional[_Address] = None,
+        deprioritized_servers: Optional[list[Server]] = None,
     ) -> Server:
         servers = self.select_servers(selector, server_selection_timeout, address)
-        if len(servers) == 1:
-            return servers[0]
-        server1, server2 = random.sample(servers, 2)
+        filtered_servers = _filter_servers(servers, deprioritized_servers)
```

Comment:
Can we rename filtered_servers->servers  so the rest of the code doesn't have to change?